### PR TITLE
[WIP] 12.0 osi version

### PR DESCRIPTION
**Pull request by dreispt**

### CODE REVIEW

File: avatax_connector/models/avalara_api.py [hunk 1]
```diff
@@ -135,14 +135,16 @@ def get_tax(self, company_code, doc_date, doc_type, partner_code, doc_code, orig
...
+            _logger.info('GetTaxrequest committing document %s (type: %s)', doc_code, doc_type)
+        else:
+            _logger.info('GetTaxRequest for document %s (type: %s)', doc_code, doc_type)
         lineslist = []
         request = self.taxSvc.factory.create('GetTaxRequest')
         request.Commit = commit
-        request.DetailLevel = 'Diagnostic'
-#        request.DetailLevel = 'Document'
+        # request.DetailLevel = 'Diagnostic'
+        request.DetailLevel = 'Document'
```
Comment (atchuthan) [inline avatax_connector/models/avalara_api.py]: @dreispt May I know why the DetailLevel is changed from Diagnostic to Document?

File: avatax_connector/models/avalara_salestax.py [hunk 2]
```diff
@@ -13,18 +11,17 @@ class exemption_code(models.Model):
     @api.multi
     @api.depends('name', 'code')
     def name_get(self):
-        return [(r.id, ('(' + r.code + ')' + ' ' + r.name)) for r in self]
+        return [(r.id, '(%s) %s' % (r.code, r.name)) for r in self]
```
Comment (atchuthan) [inline avatax_connector/models/avalara_salestax.py]: @dreispt both name and code field are not a required field, so if setup is not done properly, then customer will have (False) False
So, we can use required="1" for both the fields I think

**Review by atchuthan — COMMENTED** [review]
We are still doing QA and let you know if we have any other points to discuss

File: avatax_connector/models/avalara_salestax.py (hunk 2 above)
Comment (dreispt) [reply avatax_connector/models/avalara_salestax.py]: Good point. Or we could make if smarter and use an "if" to properly handle both cases. What do you think?

Comment (atchuthan) [reply avatax_connector/models/avalara_salestax.py]: sounds good with both approach

File: avatax_connector/models/avalara_api.py (hunk 1 above)
Comment (dreispt) [reply avatax_connector/models/avalara_api.py]: I suppose Diagnostic is not intended to run regularly on Production, and would only be for debugging purposes.
But no strong opinion about it. It is better kept as Diagnostic?

Comment (atchuthan) [reply avatax_connector/models/avalara_api.py]: @dreispt 
For one of our customers, we made a new app to add the tax based on each line and for this to work, we need DetailLevel = 'Diagnostic'
So, I would suggest to keep Diagnostic as it was in previous versions.

Comment (dreispt) [reply avatax_connector/models/avalara_api.py]: Fixed

File: avatax_connector/models/avalara_salestax.py (hunk 2 above)
Comment (dreispt) [reply avatax_connector/models/avalara_salestax.py]: Fixed

**Comment by dreispt** [comment]
@atchuthan Any more concerns? Thanks!

**Review by atchuthan — APPROVED** [review]
:+1:

File: avatax_connector/__manifest__.py [hunk 3]
```diff
@@ -1,14 +1,17 @@
-# -*- coding: utf-8 -*-
 {
     "name": "Avalara Avatax Connector",
-    'version': '12.0.1.0.0',
-    "author": "Fabrice Henrion, Sodexis Inc<dev@sodexis.com>",
-    "website": "http://www.sodexis.com",
+    "version": "12.0.2.0.0",
+    "author": "Fabrice Henrion, Sodexis Inc<dev@sodexis.com>"
```
Comment (atchuthan) [inline avatax_connector/__manifest__.py]: @dreispt 
could you please change it from "Sodexis Inc<dev@sodexis.com>" to "Sodexis"

Comment (dreispt) [reply avatax_connector/__manifest__.py]:
```suggestion
    "author": "Fabrice Henrion, Sodexis"
```

**Comment by dreispt** [comment]
I fixed the authors name.
I think we can merge this.
Of some issue is found at any time, we can always add the fix here.

Thanks!
Daniel

**Comment by atchuthan** [comment]
Great! Changes are now reflected in Odoo apps https://www.odoo.com/apps/modules/12.0/avatax_connector/

**Pull request merged**